experimental/HasseSchmidt/test/runtests.jl

```diff
@@ -0,0 +1,119 @@
+###### Stil missing ################################################ 
+# Examples for polynomial rings over fintie fields
+#
+# R, (x, y) = polynomial_ring(GF(2), ["x", "y"])
+# f = x^2 + y^2
+#
+# R, (x, y, z) = polynomial_ring(GF(3), ["x", "y", "z"])
+# f = x^2*y + z^6
+# 
+# x^2+y^2 in GF(2)[x,y] or x^2y+z^6 in GF(3)[x,y,z]
+####################################################################
+
+@testset "hasse_derivatives" begin
+  R, (x, y) = polynomial_ring(ZZ, ["x", "y"]);
+
+  result_a1 = [ [[0, 0], x^3],
+                [[1, 0], 3*x^2],
+                [[2, 0], 3*x],
+                [[3, 0], 1]]
+  @test result_a1 == hasse_derivatives(x^3)
+
+  result_a2 = [ [[0, 0], 5*x^2 + 3*y^5],
+                [[0, 1], 15*y^4],
+                [[0, 2], 30*y^3],
+                [[0, 3], 30*y^2],
+                [[0, 4], 15*y],
+                [[0, 5], 3],
+                [[1, 0], 10*x],
+                [[2, 0], 5]]
+  @test result_a2 == hasse_derivatives(5*x^2 + 3*y^5)
+
+  result_a3 = [ [[0, 0], x^2*y^3],
+                [[1, 0], 2*x*y^3],
+                [[2, 0], y^3],
+                [[0, 1], 3*x^2*y^2],
+                [[1, 1], 6*x*y^2],
+                [[2, 1], 3*y^2],
+                [[0, 2], 3*x^2*y],
+                [[1, 2], 6*x*y],
+                [[2, 2], 3*y],
+                [[0, 3], x^2],
+                [[1, 3], 2*x],
+                [[2, 3], 1]]
+  @test result_a3 == hasse_derivatives(x^2*y^3)
+
+  result_a4 = [ [[0, 0], x^4 + y^2],
+                [[1, 0], 4*x^3],
+                [[2, 0], 6*x^2],
+                [[3, 0], 4*x],
+                [[4, 0], 1],
+                [[0, 1], 2*y],
+                [[0, 2], 1]]
+  @test result_a4 == hasse_derivatives(x^4 + y^2)
+end
+
+@testset "hasse_derivatives finite fields" begin
+  R, (x, y, z) = polynomial_ring(GF(3), ["x", "y", "z"]);
+
+  result_b1 = [ [[0, 0, 0], x^2 + y^2],
+                [[1, 0, 0], 2*x],
+                [[2, 0, 0], 1],
+                [[0, 1, 0], 2*y],
+                [[0, 2, 0], 1]]
+  @test result_b1 == hasse_derivatives(x^2 + y^2)
+
+  result_b2 = [ [[0, 0, 0], x^2*y + z^6],
+                [[0, 0, 3], 2*z^3],
+                [[0, 0, 6], 1],
+                [[1, 0, 0], 2*x*y],
+                [[2, 0, 0], y],
+                [[0, 1, 0], x^2],
+                [[1, 1, 0], 2*x],
+                [[2, 1, 0], 1]]
+  @test result_b2 == hasse_derivatives(x^2*y + z^6)
+end
+
+@testset "_hasse_derivatives MPolyQuoRingElem" begin
+  R, (x, y, z) = polynomial_ring(ZZ, ["x", "y", "z"]);
+  I = ideal(R, [x^2 - 1]);
+  RQ, _ = quo(R, I);
```

Comment:
No need for those semicolons here and elsewhere in this test file
```suggestion
  R, (x, y, z) = polynomial_ring(ZZ, ["x", "y", "z"])
  I = ideal(R, [x^2 - 1])
  RQ, _ = quo(R, I)
```